File: backend/src/Models/Request/InvocationRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;
using System.Text.Json.Serialization;
using Models.Enums;

namespace Models.Request;

[ExcludeFromCodeCoverage]
public class InvocationRequest()
{
    [JsonIgnore]
    public Guid ReferenceId => Guid.TryParse(IdReference, out var guid) ? guid : Guid.Empty;

    [Required(ErrorMessage = "IdReference is required and a Guid Type.")]
    public string IdReference { get; set; } = string.Empty;
    public string? MethodName { get; set; }
    public List<ParameterSignatureRequest>? Parameters { get; set; }

    [JsonConverter(typeof(JsonStringEnumConverter))]
    [Required(ErrorMessage = "TypeReference is required and their values should be: Base, This, LocalVariable, Attribute, Parameter.")]
    public TypeReference? TypeReference { get; set; }
}

File: frontend/src/app/models/request/InvocationRequest.ts
import { ParameterSignatureRequest } from './ParameterSignatureRequest';


export class InvocationRequest {
  idReference: string = '';
  methodName: string = '';
  parameters: ParameterSignatureRequest[] = [];
  typeReference: string = ''; // 'Base' | 'This' | 'LocalVariable' | 'Attribute' | 'Parameter'
  idReturnType: string = '';
}